weatherman.py

```diff
@@ -1,47 +1,47 @@
-from Report import Report
-from Reader import Reader
-from Parser import Parser
 import argparse
 import os
 
+from reader import Reader
+from report import Report
+from parser import Parser
 
-def directory_path(path):
-    print(path)
+
+def validate_directory_path(path):
     if os.path.isdir(path):
         return path
     else:
         raise NotADirectoryError(path)
 
 
+def extract_year_and_month_from_argument(arguments):
+    return int(arguments.split("/")[0]), int(arguments.split("/")[1])
+
+
 if __name__ == '__main__':
     parser = argparse.ArgumentParser()
     parser.add_argument('-a', dest='a', type=str, help='monthly operation')
     parser.add_argument('-e', dest='e', type=int, help='yearly operation')
     parser.add_argument('-c', dest='c', type=str, help='monthly operation | generating reports')
     parser.add_argument('directory', nargs='+', action='store')
     args = parser.parse_args()
-    directory = directory_path(args.directory[0])
+    directory = validate_directory_path(args.directory[0])
     reader = Reader()
     data = reader.read_folder(directory)
     parser = Parser()
-    data = parser.record_parsing(data)
+    data = parser.parsing_records_into_dictionaries(data)
```

Comment:
As mentioned before, `data` is a vague name. Kindly make sure that you don't repeat mistakes.